### ISSUE TITLE
Decrypt Key Store Entry

### ISSUE THREAD

**Issue by jvincze84** [ansible-semaphore/semaphore#1529]
Dear All,

I would like to know if there is any way to decrypt Keystore entries?
I can find them in the databse, but they are encrypted.

Thanks a lot.

**Comment by parkjaeil0x01** [comment]
To. @jvincze84 
Encrypted key information will not be able to be decrypted...(I guess)

I'm not even sure what algorithm they use to encrypt the database. (If anyone knows, please leave a comment)
Likewise, if it uses a general encryption algorithm (such as AES), the purpose of 'encryption' will be achieved only if decryption is impossible.

[https://docs.ansible-semaphore.com/administration-guide/security](url)

Regards.

**Comment by jvincze84** [comment]
Hi,
Thans for comment.
I think Semaphore should be able to decrypt the keys, because somehow it uses them for login via ssh (for example). But I'm not sure about that.

**Comment by parkjaeil0x01** [comment]
Yeah, my last comment was based on the assumption that you didn't know the SEMAPHORE_ACCESS_KEY_ENCRYPTION key.
If you have the SEMAPHORE_ACCESS_KEY_ENCRYPTION key then, logically you should be able to decrypt it. but...
for now, I even don't know about their encryption algorithm so if you know an encryption algorithm then, you should be able to decrypt any keys from the database.

So first, we have to figure out what algorithm is used by it.

**Comment by fiftin** [comment]
Hi all, you can, but it is tricky. 

I added command `semaphore vault rekey --old-key ***`. You can decrypt all secrets in database if you remove `SEMAPHORE_ACCESS_KEY_ENCRYPTION` and `access_key_encryption` from config file. I will add docs soon.

**Comment by fiftin** [comment]
I think we can add command `semaphore vault decrypt <key_id>` to decrypt specific key. What do you think? :)

**Comment by parkjaeil0x01** [comment]
> Collaborator

What encryption algorithm is used? It's just my curiosity.
I think, it's the AES256 or something like that...

**Comment by fiftin** [comment]
@parkjaeil0x01 yes, AES256
